lib/delighted/resource.rb

```diff
@@ -11,6 +11,20 @@ def expandable_attributes
       def singleton_resource?
         !!@singleton_resource
       end
+
+      def identifier_string(id_hash)
+        raise ArgumentError, "must pass Hash" unless Hash === id_hash
+
+        id_key = id_hash.keys.detect { |k| !id_hash[k].to_s.empty? }
+        raise ArgumentError, "must pass an identifier name and value" unless id_key
+        id_value = id_hash[id_key]
```

Comment:
I think we should make this a bit stricter (and simpler?) and validate that you only pass a Hash of size 1 (so we can just `keys[0]`). The caller needs to make a decision about which identifier to use, essentially. So there's no ambiguity.